core/src/main/java/org/keycloak/representations/overrides/NoSubcomponentsComponentExportRepresentation.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.representations.overrides;
+
+import org.keycloak.common.util.MultivaluedHashMap;
+import io.fabric8.crd.generator.annotation.SchemaFrom;
+
+public class NoSubcomponentsComponentExportRepresentation {
+
+    private String id;
+    private String name;
+    private String providerId;
+    private String subType;
+    // TODO: eventually generate code for Nth levels of depth
+    // private MultivaluedHashMap<String, ComponentExportRepresentation> subComponents = new MultivaluedHashMap<>();
```

Comment:
Due to limitations in the CRD schema we should limit the recursive `subComponents` to a finite number.
What would be a reasonable number of allowed, nested, `subComponents`?
cc. @stianst @pedroigor